ir/expression.def

```diff
@@ -384,14 +384,28 @@ class ConstructorCallExpression : Expression {
 
 /// Represents a list of expressions separated by commas
 class ListExpression : Expression {
+    bool defaultInitializer = false;
+    ListExpression(Util::SourceInfo srcInfo, IR::Vector<IR::Expression>& components, bool convenientInit) : Expression(srcInfo), components(components){
```

Comment:
Can you also set defaultInitializer after a comma instead of doing it in the body of the constructor?
This is our consistent style. 
This applies to all other changes to the def files.